assets/scss/optimized-checkout.scss

```diff
@@ -171,15 +224,74 @@ a {
     color: stencilColor("optimizedCheckout-form-textColor");
 }
 
+.form-field--error {
+    .optimizedCheckout-form-label,
+    .form-inlineMessage {
+        color: stencilColor("optimizedCheckout-formField-errorColor");
+    }
+}
 //
 // Form Input
-// Background & Border Color
+// Background, Border, Shadow and Text Color
 // Styles form fields.
 // -----------------------------------------------------------------------------
 
-.optimizedCheckout-form-input {
+$optmizedCheckout-inputBoxShadow: inset 0 1px 1px stencilColor("optimizedCheckout-formField-shadowColor");
```

Comment:
Also, I think we have a specific way of naming SASS variables (i.e.: https://github.com/bigcommerce/sass-style-guide#variables). Do you want to follow that?